### PR TITLE
[crash-reporter] Fix infinite looping of privacy notice. Contributes to JB#43089

### PR DESCRIPTION
**Pull request by salmelamike**

### CODE REVIEW

File: src/sailfishui/plugin/PrivacyNotice.qml [hunk 1]
```diff
@@ -28,13 +28,17 @@ Item {
 
     property Page page
     readonly property bool pageActive: page.status === PageStatus.Active
+    property bool noticeGiven: false
 
     onPageActiveChanged: {
         if (pageActive) {
             if (PrivacySettings.privacyNoticeAccepted) {
                 pageStack.pop()
+            } else if(! noticeGiven) {
```
Comment (rainemak) [inline src/sailfishui/plugin/PrivacyNotice.qml]: Style error....

} else if (!noticeGiven) {

File: src/sailfishui/plugin/PrivacyNotice.qml [hunk 2]
```diff
@@ -28,13 +28,17 @@ Item {
 
     property Page page
     readonly property bool pageActive: page.status === PageStatus.Active
+    property bool noticeGiven: false
```
Comment (rainemak) [inline src/sailfishui/plugin/PrivacyNotice.qml]: false is default value... on qml you can guarantee that default values are what it is in docs regardless of cpu architecture. So... Maybe like.

```property bool privacyNoticeShown```

File: src/sailfishui/plugin/PrivacyNotice.qml [hunk 3]
```diff
@@ -28,13 +28,17 @@ Item {
...
     onPageActiveChanged: {
         if (pageActive) {
             if (PrivacySettings.privacyNoticeAccepted) {
                 pageStack.pop()
+            } else if(! noticeGiven) {
+                noticeGiven = true
+                pageStack.push(privacyNoticeDialog, null, PageStackAction.Immediate)
             } else {
-                pageStack.push(privacyNoticeDialog)
+                pageStack.pop()
```
Comment (rainemak) [inline src/sailfishui/plugin/PrivacyNotice.qml]: This is more like a safety guard, right?

Comment (salmelamike) [reply src/sailfishui/plugin/PrivacyNotice.qml]: This is what pops back to the settings menu if the PrivacyNotice isn't accepted.

Comment (rainemak) [reply src/sailfishui/plugin/PrivacyNotice.qml]: Yeah, checked that as well. I was surprised that this PrivacyNotice was an item not Dialog. Well... works like this as well. This crash reporter has not seen that much love lately.

**Pull request merged**